### PR TITLE
update to log4j2

### PR DESCRIPTION
**Pull request by taba90**
- update log4j dependencies to v2.
- migrate log4j.properties to log4j2.properties
- no need to update java class since slf4j is used.
- dependencyManagement for all dependencies.

### CODE REVIEW

File: src/main/resources/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+appender.console.type = Console
+appender.console.name = LogToConsole
+appender.console.layout.type = PatternLayout
+appender.console.layout.pattern = %d{ABSOLUTE} %5p %c{1}:%L - %m%n
+rootLogger.appenderRef.stdout.ref = LogToConsole
+rootLogger.appenderRef.console.ref = LogToConsole
+
+### direct messages to file hibernate.log ###
```
Comment (aaime) [inline src/main/resources/log4j2.properties:12]: Looks legit but I have to ask... has this bit been manually tested?

File: src/main/resources/log4j2.properties [hunk 2]
```diff
@@ -0,0 +1,18 @@
+rootLogger.level = INFO
```
Comment (aaime) [inline src/main/resources/log4j2.properties:1]: Also, as part of manual testing, did you try setting this to a finer level and see if you can get debug messages going? Although it does not seem the is much in terms of logging, you'd probably need to set up a proxy in order to get anything interesting in the output.

**Review by aaime — APPROVED** [review]
Looks good, just wondering about the manual testing procedures.

File: src/main/resources/log4j2.properties (hunk 1 above)
Comment (taba90) [reply src/main/resources/log4j2.properties:12]: yes it has been tested manually. See comment below.

File: src/main/resources/log4j2.properties (hunk 2 above)
Comment (taba90) [reply src/main/resources/log4j2.properties:1]: yes it has been tested manually. See comment below.

**Comment by taba90** [comment]
thanks for your review @aaime.
Here a screen shot showing log at level debug on console:
proxy-logs
This is an piece of the log when using a file appender with different layout (together with mapstore) :

```
DEBUG   2022-11-16 14:18:24.938   HTTPProxy.createHttpClient() - HTTP proxy host: null 
DEBUG   2022-11-16 14:18:24.939   HTTPProxy.createHttpClient() - HTTPS proxy host: null 
INFO   2022-11-16 14:18:24.947   HTTPProxy.createHttpClient() - HTTP Client created 
INFO   2022-11-16 14:19:07.275   Utils.buildURL() - Using default HTTPS port: 443 
INFO   2022-11-16 14:19:07.497   HTTPProxy$1.determineRoute() - HTTP proxy target host: https://gs-stable.geo-solutions.it:443 
INFO   2022-11-16 14:19:07.497   HTTPProxy$1.determineRoute() - Returning proxy route 
DEBUG   2022-11-16 14:19:07.501   HTTPProxy$1.getProxyRoute() - Setting https scheme 
DEBUG   2022-11-16 14:19:07.535   RequestAddCookies.process() - CookieSpec selected: default 
DEBUG   2022-11-16 14:19:07.589   RequestAuthCache.process() - Auth cache not set in the context 
DEBUG   2022-11-16 14:19:07.597   PoolingHttpClientConnectionManager.requestConnection() - Connection request: [route: {s}->https://gs-stable.geo-solutions.it:443][total available: 0; route allocated: 0 of 6; total allocated: 0 of 6] 
DEBUG   2022-11-16 14:19:07.680   PoolingHttpClientConnectionManager.leaseConnection() - Connection leased: [id: 0][route: {s}->https://gs-stable.geo-solutions.it:443][total available: 0; route allocated: 1 of 6; total allocated: 1 of 6] 
DEBUG   2022-11-16 14:19:07.690   MainClientExec.execute() - Opening connection {s}->https://gs-stable.geo-solutions.it:443 
DEBUG   2022-11-16 14:19:07.744   DefaultHttpClientConnectionOperator.connect() - Connecting to gs-stable.geo-solutions.it/164.132.91.198:443 
DEBUG   2022-11-16 14:19:07.745   SSLConnectionSocketFactory.connectSocket() - Connecting socket to gs-stable.geo-solutions.it/164.132.91.198:443 with timeout 0 
DEBUG   2022-11-16 14:19:07.883   SSLConnectionSocketFactory.createLayeredSocket() - Enabled protocols: [TLSv1.3, TLSv1.2, TLSv1.1, TLSv1] 
DEBUG   2022-11-16 14:19:07.883   SSLConnectionSocketFactory.createLayeredSocket() - Enabled cipher suites:....
```

**Pull request merged**